R/dv8.R

```diff
@@ -0,0 +1,153 @@
+#'
+#' @param hdsmb_path path to the matrix from hdsmb file
+#' @param sdsmb path to the matrix from sdsmb file
+#' @param hsdsmb_path path to save the merged matrices in hsdsm.dv8-dsm file
+#' @export
+dv8_hdsmb_sdsmb_to_hsdsmb <- function(hdsmb_path, sdsmb_path, hsdsmb_path){
+  # Expand paths (e.g. "~/Desktop" => "/Users/someuser/Desktop")
+  hdsmb_path <- path.expand(hdsmb_path)
+  sdsmb_path <- path.expand(sdsmb_path)
+  hsdsmb_path <- path.expand(hsdsmb_path)
+
+  # Run system2 command
+  system2('dv8-console', args=c('core:merge-matrix', '-outputFile', hsdsmb_path, hdsmb_path, sdsmb_path), stdout=FALSE, stderr=FALSE)
+}
+
+#' Convert a json cluster file, clsxj, to a binary cluster file, clsxb.
+#' An input file *-hier.json is converted to *.dv8-clsx.
+#'
+#' @param clsxj_path path to the input file project-hier.json
+#' @param clsxb_path path to clustering file (*.dv8-clsx)
+#' @export
+dv8_clsxj_to_clsxb <- function(clsxj_path, clsxb_path){
+  # Expand paths (e.g. "~/Desktop" => "/Users/someuser/Desktop")
+  clsxj_path <- path.expand(clsxj_path)
+  clsxb_path <- path.expand(clsxb_path)
+
+  # Run system2 command
+  system2('dv8-console', args=c('core:convert-cluster', '-outputFile', clsxb_path, clsxj_path), stdout=FALSE, stderr=FALSE)
+}
+
+#' Detect Architecture anti-patterns
+#'
+#' @param hsdsmb_path path to the merged matrix input file dv8-dsm
+#' @param flaws_path path to architecture folder
+#' @export
+dv8_hsdsmb_to_flaws <- function(hsdsmb_path, flaws_path){
+  # Expand paths (e.g. "~/Desktop" => "/Users/someuser/Desktop")
+  hsdsmb_path <- path.expand(hsdsmb_path)
+  flaws_path <- path.expand(flaws_path)
+
+  # Run system2 command with appropriate values for options
+  system2('dv8-console', args=c('arch-issue:arch-issue',
+                                '-uiCochange', '2',
+                                '-crossingFanIn', '4',
+                                '-crossingFanOut', '4',
+                                '-cliqueDepends', 'call,use',
+                                '-uihDepends', 'call,use',
+                                '-uihInheritance', 'extend,implement,public,private,virtual',
+                                '-outputFolder', flaws_path,
+                                hsdsmb_path),
+          stdout=FALSE, stderr=FALSE)
+}
+
+#' Parse Architecture Flaws
+#'
+#' @param file_measure_report_path path to file-measure-report.csv
+#' @param flaws_path path to architecture folder
+#' @export
+parse_dv8_architectural_flaws <- function(file_measure_report_path, flaws_path) {
+  folders_expected <- c("clique", "modularity-violation", "package-cycle", "unhealthy-inheritance")
+  folders_exist <- list.files(flaws_path)
+
+  check_folders_exist <- function(folders_expected, folder) {
+    folder_exists <- FALSE
+
+    if (folder %in% folders_expected) {
+      folder_exists <- TRUE
+    }
+
+    return(folder_exists)
+  }
+
+  convert_dsm_binary_to_json <- function(dsm_binary_file_path, json_path) {
```

Comment:
You should use your group function for this once it is implemented, and before I can accept your pull request.